Content.Server/_NF/Cargo/Systems/CargoSystem.TradeCrates.cs

```diff
@@ -0,0 +1,110 @@
+        {
+            if (_timing.CurTime <= component.ExpressDeliveryTime && isDestinated)
+                ev.Price += component.ExpressOnTimeBonus;
+            else if (_timing.CurTime > component.ExpressDeliveryTime)
+                ev.Price -= component.ExpressLatePenalty;
+        }
+        ev.Price = double.Max(0.0, ev.Price); // Ensure non-negative values.
+    }
+
+    private void OnTradeCrateInit(EntityUid uid, TradeCrateComponent component, ref ComponentInit ev)
+    {
+        // If there are no available destinations, tough luck.
+        if (_destinations.Count > 0)
+        {
+            var randomIndex = _random.Next(_destinations.Count);
+            // Better have more than one destination.
+            if (_station.GetOwningStation(uid) == _destinations[randomIndex])
+            {
+                randomIndex = (randomIndex + 1 + _random.Next(_destinations.Count - 1)) % _destinations.Count;
+            }
+            var destination = _destinations[randomIndex];
+            component.DestinationStation = destination;
+            if (TryComp<TradeCrateDestinationComponent>(destination, out var destComp))
+                _appearance.SetData(uid, TradeCrateVisuals.DestinationIcon, destComp.DestinationProto);
+        }
+
+        if (component.ExpressDeliveryDuration > TimeSpan.Zero)
+        {
+            component.ExpressDeliveryTime = _timing.CurTime + component.ExpressDeliveryDuration;
+            _appearance.SetData(uid, TradeCrateVisuals.IsPriority, true);
+
+            component.PriorityCancelToken = new CancellationTokenSource();
+            Timer.Spawn((int)component.ExpressDeliveryDuration.TotalMilliseconds,
+                () => DisableTradeCratePriority(uid, component),
+                component.PriorityCancelToken.Token);
+        }
+    }
+
+    private void OnTradeCrateRemove(EntityUid uid, TradeCrateComponent component, ref ComponentRemove ev)
+    {
+        component.PriorityCancelToken?.Cancel();
+    }
+
+    private void OnTradeCrateExamined(EntityUid uid, TradeCrateComponent component, ref ExaminedEvent ev)
+    {
+        if (!TryComp(component.DestinationStation, out MetaDataComponent? metadata))
+            return;
+
+        ev.PushMarkup(Loc.GetString("trade-crate-destination-station", ("destination", metadata.EntityName)));
+        if (component.ExpressDeliveryTime != null)
+        {
+            if (component.ExpressDeliveryTime >= _timing.CurTime)
+                ev.PushMarkup(Loc.GetString("trade-crate-priority-active"));
+            else
+                ev.PushMarkup(Loc.GetString("trade-crate-priority-inactive"));
+        }
+    }
+
+    private void DisableTradeCratePriority(EntityUid uid, TradeCrateComponent component)
+    {
+        _appearance.SetData(uid, TradeCrateVisuals.IsPriorityInactive, true);
+    }
+
+    private void OnDestinationInit(EntityUid uid, TradeCrateDestinationComponent component, ref ComponentInit ev)
```

Comment:
isnt it preferred to discard the event args in event handlers whenever they aren't used

Reply:
update: talked with pjb about it
no don't do that, it bad
c# will just create a new variable called `_`